src/ert/scheduler/scheduler.py

```diff
@@ -197,14 +200,10 @@ async def execute(
                 self._tasks[iens] = asyncio.create_task(
                     job(start, sem, self._max_submit)
                 )
-
             start.set()
-            results = await asyncio.gather(
-                *self._tasks.values(), return_exceptions=True
+            await asyncio.gather(
+                *self._tasks.values(), *background_task_list, return_exceptions=False
```

Comment:
I don't think you want to wait for background_task_list here. These are just waiting to be cancelled and will not stop by themselves.

Reply:
We have to somehow wait for them here too, in case any of them fail. Otherwise, we wont check if any of the background tasks raise exception until we exit the context.  That is probably the cause of the hanging. We are stuck here because the poll task raises exception, and we dont check for that until after the (now never-ending) jobs finish.